src/open_raw.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Tyler Neely
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use libc::{c_char, c_int};
+
+use ffi_util;
+
+use std::ffi::CString;
+use std::iter;
+use std::mem;
+use std::path::{Path, PathBuf};
+use std::ptr;
+
+use crate::{ColumnFamilyDescriptor, Error, Options};
+
+pub struct OpenRawInput<'a, T> {
+    pub(crate) options: &'a Options,
+    pub(crate) path: &'a Path,
+    pub(crate) column_families: Vec<ColumnFamilyDescriptor>,
+    pub(crate) open_descriptor: T,
+}
+
+pub struct OpenRawFFI<'a, T> {
+    pub(crate) options: *const ffi::rocksdb_options_t,
+    pub(crate) path: *const c_char,
+    pub(crate) num_column_families: c_int,
+    pub(crate) column_family_names: *mut *const c_char,
+    pub(crate) column_family_options: *mut *const ffi::rocksdb_options_t,
+    pub(crate) column_family_handles: *mut *mut ffi::rocksdb_column_family_handle_t,
+    pub(crate) open_descriptor: &'a T,
+}
+
+pub trait OpenRaw: Sized {
```

Comment:
That would require duplicating `OpenRawInput`, and `OpenRawFFI` as well. Since a `Vec` of column families can be used to represent both scenarios and since all the "raw" types are an implementation detail not exposed to users, I'd rather leave it as one trait.